### PR TITLE
Added custom vendors and eventStatus to the onConsentChange.ts

### PR DESCRIPTION
**Pull request by ioanna0**
## What does this change?
Adds the custom vendors to the onConsentChange callback

### CODE REVIEW

**Comment by mxdvl** [comment]
👍 LGTM—merge away!

File: src/onConsentChange.ts [hunk 1]
```diff
@@ -53,8 +53,29 @@ const getConsentState: () => Promise<ComparedConsentState> = () =>
...
+								);
+							} else {
+								reject(
+									new Error(
+										'An error occured while fetching getCustomVendorConsents',
+									),
+								);
+							}
+						},
+					);
```
Comment (mxdvl) [inline src/onConsentChange.ts:78]: Is there a way we could make this simpler and cleaner?

File: src/onConsentChange.ts [hunk 2]
```diff
@@ -53,8 +53,29 @@ const getConsentState: () => Promise<ComparedConsentState> = () =>
 		} else if (window.__tcfapi) {
 			// in RoW - https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md
 			window.__tcfapi('getTCData', 2, (tcfData, success) => {
-				if (success) {
-					resolve(compareState({ tcfv2: tcfData?.purpose.consents }));
+				if (success && window.__tcfapi) {
```
Comment (ioanna0) [inline src/onConsentChange.ts:56]: can be refactored to use Promise.all and remove duplication of if (window.__tcfapi)

File: src/onConsentChange.ts [hunk 3]
```diff
@@ -53,8 +53,29 @@ const getConsentState: () => Promise<ComparedConsentState> = () =>
...
+					window.__tcfapi(
+						'getCustomVendorConsents',
+						2,
```
Comment (ioanna0) [inline src/onConsentChange.ts:59]: Note to update the README to reflect the change of the new response

File: src/onConsentChange.ts (hunk 2 above)
Comment (mxdvl) [reply src/onConsentChange.ts]: Well done!

File: src/onConsentChange.ts (hunk 1 above)
Comment (mxdvl) [reply src/onConsentChange.ts]: As per @ioanna0's suggestion, this logic is now handled by promises.

**Pull request merged**